components/ILIAS/HTMLLearningModule/classes/class.ilObjFileBasedLMGUI.php

```diff
@@ -709,4 +744,68 @@ public function redrawHeaderAction(): void
     {
         $this->redrawHeaderActionObject();
     }
+
+    public function importFromUploadDir(): void
+    {
+        global $DIC;
+        if (!$this->checkPermissionBool("write", "", "htlm")) {
+            $main_tpl = $DIC->ui()->mainTemplate();
+
+            $main_tpl->setOnScreenMessage(
+                'failure',
+                sprintf(
+                    $this->lng->txt("msg_no_perm_write"),
+                ),
+                true
+            );
+            ilObjectGUI::_gotoRepositoryRoot();
+        }
+
+        $file = $this->lm_request->getString('form/input_0');
```

Comment:
This is not exactly my turf, but you should use the UI\Inputs mechanisms here top retrieve the data. Since this way it's never checks, if the value is one which is actually a file from the list (may be a security issue). additionally the internals from the Forms/Inputs could change and `'form/input_0'` will no longer be valid.

you could pack the roundtrip-modal from https://github.com/ILIAS-eLearning/ILIAS/pull/9641/files#diff-e991f544f00a77d5dc1b3b399499d56b36ea3479cfe65a418c052145ef38da92R137 in a property or method and use it's getData instead.